File: csharp/lib/Property/ChannelMaskProperty.cs
/*
 * SPDX-FileCopyrightText: 2017-2022 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

using System.Collections.Generic;
using MessagePack;

namespace SensCord
{
    /// <summary>
    /// Property for masking the channel.
    /// </summary>
    [MessagePackObject]
    public class ChannelMaskProperty : IBaseProperty
    {
        /// <summary>
        /// Property key.
        /// </summary>
        public static readonly string ConstKey = "channel_mask_property";

        /// <summary>
        /// Property key.
        /// </summary>
        [IgnoreMember]
        public string Key { get; } = ConstKey;

        /// <summary>
        /// The list of masked channel IDs.
        /// </summary>
        [Key("channels")]
        public List<long> Channels { get; set; } = new List<long>();

        /// <summary>
        /// To string.
        /// </summary>
        /// <returns>A string that represents the current object.</returns>
        public override string ToString()
        {
            return $"{this.Key}({string.Join(", ", this.Channels)})";
        }
    }
}

File: java/senscord/src/senscord_properties/ChannelMaskProperty.java
/*
 * SPDX-FileCopyrightText: 2019 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package senscord_properties;

import java.util.Arrays;
import java.util.List;
import com.sun.jna.Structure;


/**
 * Property for masking the channel.
 * 
 * @see "senscord::ChannelMaskProperty"
 */
public class ChannelMaskProperty extends Structure {
    public int count;
    public int[] channels;

    public static class ByReference extends ChannelMaskProperty implements Structure.ByReference {
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("count", "channels");
    }

    public ChannelMaskProperty() {
        super();
        channels = new int[128];
    }

    /**
     * Get masked channel count.
     * 
     * @return masked channel count
     */
    public int GetCount() {
        return count;
    }

    /**
     * Get channel masked channel ID.
     * 
     * @return masked channel ID.
     */
    public int[] GetChannels() {
        return channels;
    }
}
